src/installer/parsers.py

```diff
@@ -0,0 +1,137 @@
+"""Utilities for parsing and handling files related to a wheel (METADATA, WHEEL, RECORD)
+"""
+
+import csv
+import os
+from email.parser import FeedParser
+
+from installer._compat.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from email.message import Message
+    from typing import Iterator, Optional
+    from installer._compat.typing import FSPath
+
+__all__ = [
+    "parse_metadata_file", "parse_record_file", "Hash", "Record", "InvalidRecord",
+]
+
+
+#
+# METADATA (and WHEEL) file parsing
+#
+def parse_metadata_file(contents):
+    # type: (Text) -> Message
+    feed_parser = FeedParser()
```

Comment:
shouldn't this be a context manager?

Reply:
No... This is a weird object, that's got kinda weird naming of it's API.